module-gui/test/test-catch-text/CMakeLists.txt

```diff
@@ -6,17 +6,17 @@ add_catch2_executable(
                 ../mock/buildTextDocument.cpp
                 ../mock/multi-line-string.cpp
                 ../mock/InitializedFontManager.cpp
-                test-gui-Text.cpp
-                test-gui-TextFixedSize.cpp
                 test-gui-TextBlock.cpp
                 test-gui-TextBlockCursor.cpp
                 test-gui-TextDocument.cpp
                 test-gui-MultiTextLine.cpp
                 test-gui-TextParse.cpp
-                test-gui-Font.cpp
-                test-gui-TextLinesCursor.cpp
-		        test-gui-TextInLineCursor.cpp
-		INCLUDE
+                "$<$<STREQUAL:${ASSETS},Proprietary>:test-gui-Text.cpp>"
```

Comment:
I believe you missed `ASSETS_TYPE`.

This is a place where you should rather just move files to ASSET_TYPE dependent catalog and just source them from there. Then you will have one include of each - each ASSET_TYPE dependent one.